src/miniscript/astelem.rs

```diff
@@ -993,7 +994,14 @@ impl<P: ToPublicKey> AstElem<P> {
             AstElem::Thresh(k, ref subs) |
             AstElem::ThreshV(k, ref subs) => 1 +
                 script_num_size(k) +
-                subs.iter().map(|s| s.script_size()).sum::<usize>()
+                subs.iter().enumerate().map(|(n, s)|
+                    if n == 0 {
+                        s.script_size()
+                    }
+                    else {
+                        1 + s.script_size()
+                    }
+                    ).sum::<usize>()
```

Comment:
I think we can simplify this to simply adding `subs.len() - 1` to the old code (and maybe an `assert!(!subs.is_empty())` which I think/hope we enforce at parse time).

Reply:
Fixed.
I didn't add the assertion since I thought this is not the right place to do.

Reply:
The code will panic even without the assertion, if `subs.len()` is 0 and you try to subtract one.

Reply:
I agree it's not really appropriate to assert here, but if you don't want to, you need to add an explicit `subs.is_empty()` check. (Your original code did the right thing actually :})

Reply:
Well, then I guess just assert here.
empty `subs` are something that shouldn't happen. so failing fast seems better than returning  nonsense value.(IMHO)